app/models/post.rb

```diff
@@ -1,17 +1,21 @@
-class Post < ActiveRecord::Base
-  belongs_to :user, class_name: 'User', foreign_key: :user_id
-  has_many :likes
-  has_many :comments
+class Post < ApplicationRecord
+  validates :title, presence: true, length: { maximum: 250, too_long: '%<count>s characters is the maximum allowed' }
+  validates :comments_counter, :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
-  def five_last_comments
-    comments.order('created_at Desc').limit(5)
-  end
+  belongs_to :user, counter_cache: :posts_counter
+  has_many :comments, dependent: :destroy
+  has_many :likes, counter_cache: :likes_counter
+  has_many :users, through: :likes, counter_cache: :likes_counter
 
-  after_save :update_posts_counter
+  def recent_comments(limit = 5)
+    comments.last(limit)
+  end
 
-  private
+  def liked?(user_id)
+    likes.where(user_id:).exists?
```

Comment:
- [ ] Kindly note that I am getting a syntax error at this line when showing a user. Kindly correct this syntax error.

```.rb 
  def liked?(user_id)
    likes.where(user_id:).exists?
  end
```
![image](https://user-images.githubusercontent.com/65068771/157906116-f7053321-bf77-4e95-afd3-bd3a25f1fc7e.png)

Reply:
This issue has been resolved